packages/core/src/common/uri-command-handler.ts

```diff
@@ -35,24 +35,24 @@ export interface UriCommandHandler<T extends MaybeArray<URI>> extends CommandHan
 /**
  * Handler for a single URI-based selection.
  */
-export interface SingleUriCommandHandler extends UriCommandHandler<URI> {
+export interface SingleUriCommandHandler extends UriCommandHandler<URI, any> {
 
 }
 
 /**
  * Handler for multiple URIs.
  */
-export interface MultiUriCommandHandler extends UriCommandHandler<URI[]> {
+export interface MultiUriCommandHandler extends UriCommandHandler<URI[], any> {
 
 }
 
-export class UriAwareCommandHandler<T extends MaybeArray<URI>> implements UriCommandHandler<T> {
+export class UriAwareCommandHandler<T extends MaybeArray<URI>, R = any> implements UriCommandHandler<T, R | undefined> {
```

Comment:
See my comment on the `plugin-vscode-commands` change. These handlers can be called with anything. If it's a URI / URI[], then it will be used as the first arg. Otherwise a new first arg will be supplied by the command service and the supplied args will be shifted.

Reply:
Honestly this `uri-command-handler` file is a pain to type... What was missing was a clear understanding of the expected behavior, thanks for clearing this out. If you see a way to properly type this then feel free to push a commit on top of this branch.